### DISCUSSION TITLE
Use x-if and x-teleport together

### DISCUSSION THREAD

**Discussion by Babelfisch** [alpinejs/alpine#3684] [1. Help] [answered]
Is there a way to use `x-if` and `x-teleport` together?

This always shows the template and ignores the `x-if` directive:
```html
<template x-data="{ visible: true }" x-if="visible" x-teleport="body">
…
</template
```

**Comment by ekwoka** [comment, marked as answer]
they, like x-for all operate with the template independently.

You can put the x-teleport on a template inside the x-if.

Or the opposite as well, x-if inside the x-teleport.
